### ISSUE TITLE
Improve server packet decoding performance

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#1407]
1. Don't decode packets from unconnected players: https://github.com/ddnet/ddnet/commit/9089c2b35ad19d7b80b23e6583adb2662ca57b83
Didn't work for some, might need protocol changes. Anyone know if compressed packets are possible for unconnected players? Maybe because of the connection protocol?

2. recvmmsg: https://github.com/ddnet/ddnet/commit/de5fe64be582c39906644dd4c60f2b032858ecab
Didn't offer additional performance, probably need 1. as well.

3. Multithreaded packet parsing and decoding
Basically spawn more threads and put the parsed results in a queue for the main thread to read from.

4. Upgrade to a server with more CPU cores and stable 1-10 gbit/s network.

This should be the way to go to prevent many application specific DoS attacks. Comments, ideas, anyone want to try out something?

**Comment by def-** [comment]
We might also consider breaking backwards-compatibility with Vanilla 0.6 since it's not supported anymore anyway.
